### ISSUE TITLE
Upgarde of TLS 1.2 to TLS 1.3 for version SwiftMQ version 9.3.1 failing

### ISSUE THREAD

**Issue by surojit214** [iitsoftware/swiftmq-ce#100]
Have been using SwiftMQ 9.3.1 with TLS 1.2. Recently updated it to TLS 1.3 and the inter router network seems to have been broken across routers after upgrade to 1.3. Things have been working fine in the past with a custom implementation to "ssl-engine-factory-class" at the network swiftlet layer. Even now after upgrade to TLS 1.3, the Handshake seems to be working fine, but getting an exception: "javax.net.ssl.SSLProtocolException: Receiving application data before handshake complete". And the issue that is being observed is that the router and not interconnected, even with multiple <connector> configuration being available at the "routing" swiftlet level. Any help is appreciated.

**Comment by edgebroker** [comment]
did you try with the latest release 12.5.4? 

9.3.1 is too old.

**Comment by surojit214** [comment]
True, but we don't have the option to upgrade it to latest, and have to stick to 9.3.1 only

**Comment by edgebroker** [comment]
What about using an SSH tunnel instead?

**Comment by surojit214** [comment]
Like mentioned above, we have overridden the "ssl-engine-factory-class" at the network swiftlet layer at it was working fine for TLS 1.2. After upgrade to TLS 1.3, what we are noticing is that the inter router network connection gets broken. We have verified the behaviour by enabling JMS and looking into the "sys$net -> usages -> connections" which comes as blank, even when multiple connectors are configured. Any help or suggestion is really appreciated. Upgrade to latest version is not an option. Would like to hear any suggestion from code perspective that can be looked into to find the root cause of the issue and to fix it.

**Comment by iitsoftware** [comment]
AFAIK, you are using SwiftMQ UR/HA but not CE. Only CE is open source so you can't fix UR/HA without violating our license.

Never heard that someone has to stay at a particular release. If so, then accept the limits of that release or upgrade. 

Issue closed.
